### ISSUE TITLE
Suggestions on post validation line & column information?

### ISSUE THREAD

**Issue by ascertrobw** [networknt/json-schema-validator#340]
One aspect we are really battling with.

How to log line/column information to a user for a YAML document that passes schema validation, but then fails semantic checks on the content of the JSON node tree?

Maybe we're missing something obvious here - but so far we have a multi step process:

1. Loading the YAML file/stream into JSON and then passing the loaded JSON tree to the factory for validation
2. Application code for processing the validated Node tree

After step 1 we have effectively lost the original source file information, so if we detect any semantic errors we can't easily report them back to the user with line/col information against the original file.

I doubt this is feasible, but I'm wondering if there couldn't be an option to include meta information in the processed JSON Node tree such as the original source positions of various nodes, which would then allow such error reporting.

**Comment by ascertrobw** [comment]
Closing this issue as we found a way to handle this. 

If it's of any interest, it wasn't that hard, involving a few fairly simple steps.

- Create a custom subclass of JsonNodeFactory that had access to the YAMLParser, where "token" and "current" file location information is available
- Create custom subclass of the various JsonNodes (TextNode, NumericNode, ObjectNode etc etc) which can be instantiated with this location information
- When asked to create a new JsonNode, the custom JsonNodeFactory returns one of the new subclassed node types, with the YAMLParser file location information embedded within it

**Comment by stevehu** [comment]
@ascertrobw Your workaround makes perfect sense. I am wondering if you want to contribute to the library so that other users can enjoy it. Thanks.

**Comment by ascertrobw** [comment]
> 
> 
> @ascertrobw Your workaround makes perfect sense. I am wondering if you want to contribute to the library so that other users can enjoy it. Thanks.

Not sure how generally applicable it will be - but happy to share the code I used. Is there an "examples" or "library" section somewhere for that, or should I just ZIP and post here?

**Comment by ascertrobw** [comment]
On a similar topic - I'm not seeing any way to get line/col information out of a ValidationMessage when a document fails validation against a schema. Still digging to see if any is recorded anywhere

**Comment by stevehu** [comment]
@ascertrobw I just sent an invite to join the team. You can create a feature branch and add an example folder to show us how it works. If we can find a way to generalize it, then we merge into the library. Otherwise, it will live in the example folder for users to adopt.

**Comment by ascertrobw** [comment]
> 
> 
> @ascertrobw I just sent an invite to join the team. You can create a feature branch and add an example folder to show us how it works. If we can find a way to generalize it, then we merge into the library. Otherwise, it will live in the example folder for users to adopt.

Cool, thanks, will do. I'm just chasing down the "line numbers in ValidationMessage" aspect, see if I can create an example for that also. Shoild hopefull be able to carve out some time to create standalone examples in the next week or two once I'm done and have got my current work off my desk.

**Comment by ascertrobw** [comment]
> 
> 
> @ascertrobw I just sent an invite to join the team. You can create a feature branch and add an example folder to show us how it works. If we can find a way to generalize it, then we merge into the library. Otherwise, it will live in the example folder for users to adopt.

A tad short of time this week to create a full worked exampled - so I've created a .MD instead which describes the scenarios and solutions
